--- a/kernel/drv/oss_hdaudio/.devices
+++ b/kernel/drv/oss_hdaudio/.devices
@@ -13,6 +13,7 @@
 oss_hdaudio	pci8086,1e20	Intel High Definition Audio (PPT)
 oss_hdaudio	pci8086,811b	Intel High Definition Audio (SCH)
 oss_hdaudio	pci8086,8c20	Intel High Definition Audio (PCH_C)
+oss_hdaudio	pci8086,a0c8    Tiger Lake-LP Smart Sound Technology Audio Controller
 oss_hdaudio	pci10de,26c	Nvidia High Definition Audio (MCP51)
 oss_hdaudio	pci10de,371	Nvidia High Definition Audio (MCP55)
 oss_hdaudio	pci10de,3e4	Nvidia High Definition Audio (MCP61)
--- a/kernel/drv/oss_hdaudio/oss_hdaudio.c
+++ b/kernel/drv/oss_hdaudio/oss_hdaudio.c
@@ -38,6 +38,7 @@
 #define INTEL_DEVICE_PPT        0x1e20
 #define INTEL_DEVICE_SCH        0x811b
 #define INTEL_DEVICE_PCH_C      0x8c20
+#define INTEL_DEVICE_TGL        0xa0c8
 
 #define NVIDIA_VENDOR_ID        0x10de
 #define NVIDIA_DEVICE_MCP51     0x026c
@@ -1824,6 +1825,7 @@
     case INTEL_DEVICE_PCH_C:
       devc->chip_name = "Intel HD Audio";
       break;
+    case INTEL_DEVICE_TGL:
 
     case NVIDIA_DEVICE_MCP51:
     case NVIDIA_DEVICE_MCP55:
