diff --git a/meson.build b/meson.build
index 360db8fa..c4e8b0d4 100644
--- a/meson.build
+++ b/meson.build
@@ -298,14 +298,17 @@ if tic.found()
     output: 'foot.info.preprocessed',
     configuration: conf_data,
   )
-  custom_target(
-    'terminfo',
-    output: get_option('default-terminfo')[0],
-    input: preprocessed,
-    command: [tic, '-x', '-o', '@OUTDIR@', '-e', '@0@,@0@-direct'.format(get_option('default-terminfo')), '@INPUT@'],
-    install: true,
-    install_dir: terminfo_install_location
-  )
+  foreach ti : [get_option('default-terminfo'),
+                get_option('default-terminfo') + '-direct']
+    custom_target(
+      ti + '-terminfo',
+      output: ti + '-terminfo',
+      input: preprocessed,
+      command: [tic, '-x', '-o', '@OUTPUT@', '@INPUT@', ti],
+      install: true,
+      install_dir: terminfo_install_location + '/' + get_option('default-terminfo')[0]
+    )
+  endforeach
 endif
 
 subdir('completions')
