diff --git a/color.c b/color.c
index c510b96..b2c7a2e 100644
--- a/color.c
+++ b/color.c
@@ -133,9 +133,8 @@ static const struct mapping_t ComposeFields[] =
 
 #define COLOR_QUOTE_INIT	8
 
-#ifdef NCURSES_VERSION
 #define ATTR_MASK (A_ATTRIBUTES ^ A_COLOR)
-#elif defined (USE_SLANG_CURSES)
+#if defined (USE_SLANG_CURSES)
 #define ATTR_MASK (~(unsigned int)A_NORMAL ^ (A_CHARTEXT | A_UNUSED | A_COLOR))
 #endif
 
